odoo/models.py

```diff
@@ -6460,6 +6460,19 @@ def diff(self, other, force=False):
         # and depends on line.a, but line.b is not in the form view)
         record._update_cache(changed_values, validate=False)
 
+        # update the cache of the definition record to make the onchange
+        # work on properties fields
+        for name in changed_values:
+            field = self._fields.get(name)
+            if field and field.type == "properties":
+                # update the cache of the definition record
+                definitions = copy.deepcopy(changed_values[name])
+                for definition in definitions:
+                    definition.pop('value', None)
+                    definition.pop('definition_changed', None)
+                definitions = [definition for definition in definitions if not definition.get('definition_deleted')]
+                record[field.definition_record]._update_cache({field.definition_record_field: definitions})
```

Comment:
The `record` is a _new_ record, or a "virtualized version" of a _real_ record.
The method `onchange` does not modify the real record, but only its virtualized version.
This statement updates the cache of a _real_ record, and therefore introduces cache inconsistencies.

Reply:
so I should create a "new record" for the parent as well ? 
(I pushed what I have in mind in a second commit)